src/posix/main.c

```diff
@@ -295,10 +295,11 @@ static otInstance *InitInstance(PosixConfig *aConfig)
     syslog(LOG_INFO, "Thread version: %hu", otThreadGetVersion());
     IgnoreError(otLoggingSetLevel(aConfig->mLogLevel));
 
-    instance = otSysInit(&aConfig->mPlatformConfig);
+    otSysInit(&aConfig->mPlatformConfig);
     syslog(LOG_INFO, "Thread interface: %s", otSysGetThreadNetifName());
 
-    atexit(otSysDeinit);
+    instance = otSysGetInstance();
+    assert(instance != NULL);
```

Comment:
Will it be a problem that `InitInstance` returns a NULL silently in release build?

Reply:
I think we expect have already catch bugs like this in debug build. 

This is an problem for all `assert` and `OT_ASSERT` in OpenThread code.

Reply:
Changed to VerifyOrDie